### PR TITLE
Added dropdown menu to view

### LINKED ISSUE #13: Design responsive application that will be rendered
User should be able to interact with widget and design should be clean (list of documents, drop down to assign reviewers). Hard code a list of reviewers temporarily. A different issue will populate the drop down.

### PR DESCRIPTION
**Pull request by mtandon18**
Resolves #13 

**Description**
This PR adds a frontend view so that reviewers can be assigned to documents.

**Code Changes**
I have added a dropdown menu using the `select` tag in a new HTML template. The options are supposed to be user names and will be populated using context. This is a useful [link](https://stackoverflow.com/questions/41155743/django-templates-and-drop-down-list) demonstrating how context will be built in. The new html templates are tethered to the view by the `SingleObjectDropdownListView` class in `generics.py`. `ReviewerManagementView` is a child class of this generic which adds context and should respond to frontend events. @Yu-Zhou-Wang and @SantianLin all you code should theoretically live in the `ReviewerManagementView` class.

**Testing**
Unfortunately, I am unable to upload documents and cannot test at this time. I would really appreciate if any of the reviewers could confirm that they see a dropdown menu when visiting this view. This cannot be done immediately due to the dependency mentioned below.

**Dependencies**
The code for linking to this view is currently in this open [PR](https://github.com/CMU-313/fall-2021-hw2-crabs-adjust-humidity/pull/17) so until that is addressed, this PR cannot be tested.

**Next Steps**
@Yu-Zhou-Wang 
As described [here](https://www.w3schools.com/tags/tag_option.asp), The `option` tag has a value attribute which returns some state to the backend. I have arbitrarily set it as the user's name but you may need to change it based on the arguments your API call will need. You also need to look into how to capture that returned value to pass it into the API call.

@SantianLin 
I have left a comment which is where your API call needs to go. If user's have a "name" field, the code should just work and the names should appear in the dropdown. If not, message me, and I can help resolve the problem.

### CODE REVIEW

File: mayan/apps/appearance/templates/appearance/generic_list_items_dropdown_subtemplate.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                                                            <a href="{{ object.get_absolute_url }}">{{ object }}</a>
+                                                        {% else %}
+                                                            {{ object }}
+                                                        {% endif %}
+                                                    {% else %}
+                                                        {% navigation_get_source_columns source=object only_identifier=True as source_column %}
+                                                        {% navigation_source_column_resolve column=source_column as column_value %}
+                                                        {{ column_value }}
+                                                    {% endif %}
+                                                    <select>
```
Comment (mtandon18) [inline mayan/apps/appearance/templates/appearance/generic_list_items_dropdown_subtemplate.html:41]: Lines 41-46 are where the dropdown lives. @Yu-Zhou-Wang the value attribute is how you will find out which reviewer was selected. You will likely also need to extract the document file name so you can modify it's ACL with reference to this user.

File: mayan/apps/documents/views/reviewer_management_views.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                message=_(
+                    'Error retrieving document list: %(exception)s.'
+                ) % {
+                    'exception': exception
+                }, request=self.request
+            )
+            self.object_list = Document.valid.none()
+            context = super().get_context_data(**kwargs)
+            
+        # context['users'] = ADD API CALL FOR LIST OF USERS HERE
```
Comment (mtandon18) [inline mayan/apps/documents/views/reviewer_management_views.py]: @SantianLin  This is where your API call needs to go

**Comment by SantianLin** [comment]
creating a new template-model-view as the StackOverflow showed, on top of Maggie's link though

**Review by SantianLin — APPROVED** [review]
matches with my own code, neat

**Review by Yu-Zhou-Wang — APPROVED** [review]

**Pull request merged**